### PR TITLE
Support event streams that are shared between two operations.

### PR DESCRIPTION
**Pull request by adamthom-amzn**
## Description
Event streams do not have to be 1:1 with operations, but the generation of event stream model types selects the first operation linked with an event stream when generating.  This change adds support for all associated operations to event stream interfaces and event types.

## Motivation and Context
This brings event stream client generation closer to the models allowed by the event stream specification.

## Testing
I added a generation test for a model that shares an event stream between two operations, and performed an end-to-end test offline with a test service that reuses event stream structures.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  private Person(BuilderImpl builder) {
+    this.name = builder.name;
+    this.birthday = builder.birthday;
+  }
+
+  /**
+   * Returns the value of the Name property for this object.
+   * @return The value of the Name property for this object. */
+  public String name() {
+    return name;}
```
Comment (millems) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java:74]: Formatting seems off. Did the generator do this? Does this compile?

Comment (adamthom-amzn) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java:74]: Yeah, I just copy and pasted the output from the generator and reviewed it to make sure it generated the methods I wanted.  IDEA only complains about missing types when I move it into a source root.

Comment (millems) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java:74]: Can we run a quick formatter on it? The comparison doesn't care about whitespace, I don't think.

Comment (adamthom-amzn) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java]: Ran IDEA's formatter on all of the newly added generated classes.

Comment (adamthom-amzn) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java]: And broke the tests.  Don't merge this yet.

Comment (adamthom-amzn) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/sharedstream/person.java]: IDEA changed

```
/*
 */
```
to

```
/*
 *
 */
```

which, of course, is not whitespace-only.  I pushed an update, should be good as soon as the automated tests don't call me a liar.

**Pull request merged**